src/askvm/resources/list/__tests__/map.test.ts

```diff
@@ -44,3 +44,11 @@ test('map', async () => {
     )
   ).resolves.toStrictEqual(namesWithPrefix);
 });
+
+test('map should not mutate the array', async () => {
+  const array = [1, 2, 3, 4];
+
+  const anotherArray = await map.resolver(array, (item: number) => item + 1);
+
+  expect(array !== anotherArray).toEqual(true);
```

Comment:
Let's make use of built-in assertion methods

```suggestion
expect(array).not.toBe(anotherArray);
```

Can you also add a check to see if the array was mapped correctly and that the previous array is left intact?